src/main/java/com/team766/framework3/ContextImpl.java

```diff
@@ -405,96 +345,119 @@ public void yield() {
     }
 
     @Override
-    public LaunchedContext startAsync(final RunnableWithContext func) {
-        return new ContextImpl(func, this);
+    public LaunchedContext startAsync(final Procedure procedure) {
+        checkProcedureReservationsDisjoint(procedure);
+        var context = new ContextImpl(procedure);
+        context.schedule();
+        return context;
+    }
+
+    @Override
+    public void runSync(final Procedure procedure) {
```

Comment:
I don't think Java offers a way to prevent that. But I think it's ok because subsystem reservation issues would still be caught by checkContextOwnership. Could also catch this with static analysis

Reply:
> the reservations wouldn't necessarily be a subset of the parent/top-level Procedure's reservations

If people follow the constructor dependency injection / `this.mech = reserve(mech);` pattern, then they should be. Agreed that at some point we're just relying on convention, but it's several layers of convention that they'd have to break to make this go wrong.

We can definitely add static analysis for a lot of this.